### DISCUSSION TITLE
Decluttering repositories

### DISCUSSION THREAD

**Discussion by harivishvanath7** [community/community#136494] [Repositories]
I currently have 33 repositories in my account, and I'm looking to declutter by consolidating these into a single new repository. My goal is to move (cut-paste) the contents of these repos into the new one while preserving my contribution history (the green spots on my contribution graph).

I attempted to do this by creating a new repository, moving the contents of the old repos, and then deleting the originals. Unfortunately, I noticed that my contribution history was affected, and some contributions were lost.

Could anyone guide me on the correct process to combine multiple repositories into one without losing my contributions? I want to ensure that all my commits and contribution history remain intact after the move.

Any help or suggestions would be greatly appreciated!

Thanks in advance!

**Comment by davevad93** [comment]
Hi @harivishvanath7 ,

Take a look at my response in your discussion #136493 , hope it helps.

**Comment by Thesirix** [comment]
Hi,

To merge multiple repositories while preserving your contribution history, you can follow these steps:

Clone the existing repositories: Start by cloning all the repositories you want to merge into your local machine.

Merge Git histories: For each repository, add it as a remote repository in your new target repository. Follow these steps for each repository:

Navigate to the new repository: **cd /path/to/new-repo**
Add the existing repository as a remote: **git remote add repo1 /path/to/repo1**
Fetch the branches and history: **git fetch repo1**
Merge a specific branch into the new repository: **git merge repo1/main --allow-unrelated-histories**
Repeat these steps for each repository.
Check the commit history: After merging all repositories, ensure that the commit history is preserved by using the **git log** command.

Push to the remote repository: Once you are satisfied with the result, push the new repository with the combined history to GitHub: **git push origin main.**

Delete the old repositories: If everything is in order and the history is preserved, you can delete the old repositories.

By following this method, you should retain the full history of your contributions since each commit will be transferred to the new repository.

i hope i help :)
